File: .clasp-loops/swarm-ready-full-goal-live/baseline-workspace/benchmarks/tasks/ts-agent-escalation/repo/src/shared/agent.ts
export type AgentAction = "reply" | "ask_followup";

export interface AgentDecision {
  action: AgentAction;
  summary: string;
  confidence: number;
}

function isRecord(value: unknown): value is Record<string, unknown> {
  return typeof value === "object" && value !== null;
}

export function isAgentDecision(value: unknown): value is AgentDecision {
  if (!isRecord(value)) {
    return false;
  }

  return (
    (value.action === "reply" || value.action === "ask_followup") &&
    typeof value.summary === "string" &&
    typeof value.confidence === "number" &&
    value.confidence >= 0 &&
    value.confidence <= 1
  );
}

File: .clasp-loops/swarm-readiness-native/baseline-workspace/benchmarks/tasks/py-agent-escalation/repo/src/shared/agent.py
from typing import Literal, TypedDict


AgentAction = Literal["reply", "ask_followup"]


class AgentDecision(TypedDict):
    action: AgentAction
    summary: str
    confidence: float


def is_agent_decision(value: object) -> bool:
    if not isinstance(value, dict):
        return False

    action = value.get("action")
    summary = value.get("summary")
    confidence = value.get("confidence")

    return (
        action in ("reply", "ask_followup")
        and isinstance(summary, str)
        and isinstance(confidence, (int, float))
        and 0 <= confidence <= 1
    )
